Purchases/Public/RCPurchases.m

```diff
@@ -219,32 +209,53 @@ - (void)cachePurchaserInfo:(RCPurchaserInfo *)info {
     }
 }
 
-- (void)updateCaches
+- (void)updateCaches {
+    [self updateCachesWithCompletionBlock:nil];
+}
+
+- (void)updateCachesWithCompletionBlock:(RCReceivePurchaserInfoBlock _Nullable)completion
 {
     NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
     if (self.cachesLastUpdated != nil && timeSinceLastCheck < 60.) {
-        [self readPurchaserInfoFromCache];
         return;
     }
 
     self.cachesLastUpdated = [NSDate date];
 
     [self.backend getSubscriberDataWithAppUserID:self.appUserID
-                                      completion:^(RCPurchaserInfo * _Nullable info,
-                                                   NSError * _Nullable error) {
-        if (error == nil) {
-            [self handleUpdatedPurchaserInfo:info error:nil];
-        } else {
-            self.cachesLastUpdated = nil;
-        }
-    }];
-
-    [self getEntitlements:^(NSDictionary<NSString *,RCEntitlement *> *entitlements) {}];
-}
-
--(NSDictionary<NSString *, RCEntitlement *> * _Nullable)entitlements
-{
-    return self.cachedEntitlements;
+                                 completion:^(RCPurchaserInfo * _Nullable info,
+                                              NSError * _Nullable error) {
+                                     if (error == nil) {
+                                         [self handleUpdatedPurchaserInfo:info error:nil];
+                                     } else {
+                                         self.cachesLastUpdated = nil;
+                                     }
+                                     
+                                     if (completion) {
+                                         [self dispatch:^{
+                                             completion(info, error);
+                                         }];
+                                     }
+                                 }];
+
+    [self getEntitlements:nil];
+}
+
+- (void)purchaserInfoWithCompletionBlock:(RCReceivePurchaserInfoBlock)completion
+{
+    RCPurchaserInfo *info = [self readPurchaserInfoFromCache];
```

Comment:
We are not checking if the caches are valid in this case. Is this intended? When reopening the app after a while and without network connectivity, this will return a possibly old version of PurchaserInfo (which it's actually probably not an issue)

Reply:
Ok yea, I rethought this and just pushed a change to the behavior.

```
RCPurchaserInfo *infoFromCache = [self readPurchaserInfoFromCache];
    if (infoFromCache) {
        CALL_IF_SET(completion, infoFromCache, nil);
        if ([self isCacheStale]) {
            [self updateCaches];
        }
    } else {
        [self updateCachesWithCompletionBlock:completion];
    }
```
If we have a cached info, send it right away, then, if the cache is stale, fire off an update.

If we don't have a cached info, request one right away.